spyder/widgets/tests/test_pathmanager.py

```diff
@@ -114,5 +110,187 @@ def test_synchronize_with_PYTHONPATH(pathmanager, mocker):
     set_user_env(listdict2envdict(env))
 
 
+@pytest.mark.parametrize('pathmanager',
+                         [(sys.path[:-10], sys.path[-10:], ())],
+                         indirect=True)
+def test_invalid_directories(qtbot, pathmanager):
+    """Check [site/dist]-packages are invalid paths."""
+    if os.name == 'nt':
+        paths = ['/lib/site-packages/foo',
+                 '/lib/dist-packages/foo']
+    else:
+        paths = ['/lib/python3.6/site-packages/foo',
+                 '/lib/python3.6/dist-packages/foo']
+
+    def interact_message_box():
+        child = pathmanager.findChild(QMessageBox)
+        qtbot.keyPress(child, Qt.Key_Enter)
+
+    for path in paths:
+        timer = QTimer()
+        timer.setSingleShot(True)
+        timer.timeout.connect(interact_message_box)
+        timer.start(300)
+        assert not pathmanager.check_path(path)
+        pathmanager.add_path(path)
+
+
+@pytest.mark.parametrize('pathmanager',
+                         [(('/spam', '/bar'), ('/foo', ), ())],
+                         indirect=True)
+def test_remove_no(qtbot, pathmanager):
```

Comment:
Please add a docstring and improve the test name because I don't understand what it does.